examples/react-hooks/next/pages/search/[category].tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import Head from 'next/head';
+import { GetServerSideProps } from 'next';
+import { renderToString } from 'react-dom/server';
+import algoliasearch from 'algoliasearch/lite';
+import { Hit as AlgoliaHit } from 'instantsearch.js';
+import {
+  DynamicWidgets,
+  InstantSearch,
+  Hits,
+  Highlight,
+  RefinementList,
+  SearchBox,
+  InstantSearchServerState,
+  InstantSearchSSRProvider,
+  useMenu,
+} from 'react-instantsearch-hooks-web';
+import { getServerState } from 'react-instantsearch-hooks-server';
+import type { UiState } from 'instantsearch.js';
+import { Panel } from '../../components/Panel';
+import Menu from '../../components/Menu';
+import { MenuConnectorParams } from 'instantsearch.js/es/connectors/menu/connectMenu';
+
+import { createInstantSearchRouterNext } from 'react-instantsearch-hooks-router-nextjs';
+import singletonRouter from 'next/router';
+
+const client = algoliasearch('latency', '6be0576ff61c053d5f9a3225e2a90f76');
+
+function VirtualMenu(props: MenuConnectorParams) {
+  useMenu(props);
+  return null;
+}
+
+type HitProps = {
+  hit: AlgoliaHit<{
+    name: string;
+    price: number;
+  }>;
+};
+
+function Hit({ hit }: HitProps) {
+  return (
+    <>
+      <Highlight hit={hit} attribute="name" className="Hit-label" />
+      <span className="Hit-price">${hit.price}</span>
+    </>
+  );
+}
+
+type CategoryPageProps = {
+  serverState?: InstantSearchServerState;
+  url?: string;
+  category?: string;
+};
+
+function FallbackComponent({ attribute }: { attribute: string }) {
+  return attribute.toLowerCase().includes('categor') ? null : (
+    <Panel header={attribute}>
+      <RefinementList attribute={attribute} />
+    </Panel>
+  );
+}
+
+type RouteState = {
+  category?: string;
+  brand?: string;
+};
+
+export default function CategoryPage({ serverState, url }: CategoryPageProps) {
+  return (
+    <InstantSearchSSRProvider {...serverState}>
+      <Head>
+        <title>React InstantSearch Hooks - Next.js</title>
+      </Head>
+
+      <InstantSearch<UiState, RouteState>
+        searchClient={client}
+        indexName="instant_search"
+        routing={{
+          router: createInstantSearchRouterNext({
+            singletonRouter,
+            serverUrl: url,
+            routerOptions: {
+              createURL({ routeState, location, qsModule }) {
+                const { category, ...queryParameters } = routeState;
+                const { origin, pathname } = location;
+
+                return `${origin}${pathname}${qsModule.stringify(
```

Comment:
it's good for a demo, but do we want to encourage the main next demo have two different InstantSearch instances?

Reply:
Yeah I guess I should make a separate demo called `next-routing` maybe ?